pkg/api/v20191231preview/openshiftcluster_validatestatic.go

```diff
@@ -134,9 +134,13 @@ func (sv *openShiftClusterStaticValidator) validateClusterProfile(path string, c
 		strings.ContainsRune(strings.TrimSuffix(cp.Domain, "."+sv.domain), '.') {
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".domain", "The provided domain '%s' is invalid.", cp.Domain)
 	}
-	if isCreate && cp.Version != version.OpenShiftVersion {
```

Comment:
This Commit (Ability to support multi-version upgrades) requires review. 
/cc @jim-minter @m1kola

Reply:
@mjudeikis you're going to have to talk me through this - I don't understand these changes to pkg/api.  Elsewhere we default Version in if it is unspecified; the changes here are guarded by isCreate ... what's the meaning of them?